bin/msstats.R

```diff
@@ -1,76 +1,132 @@
 #!/usr/bin/env Rscript
-library(dplyr)
-library(tidyr)
-library(magrittr)
-library(MSstats)
+library(rlang)
+library(dplyr, warn.conflicts = FALSE)
+library(tidyr, warn.conflicts = FALSE)
+library(stringr, warn.conflicts = FALSE)
+library(magrittr, warn.conflicts = FALSE)
+library(MSstats, warn.conflicts = FALSE)
 
 
 # Convert EncyclopeDIA results to a format for MSstats
 encyclopediaToMsstats <- function(peptides_txt) {
   id_vars <- c("Peptide", "Protein", "numFragments")
-  
-  df <- read.table(peptides_txt, sep = '\t', header = TRUE) %>% 
-    pivot_longer(names(.)[!names(.) %in% id_vars], 
+
+  df <- read.table(peptides_txt,
+                   sep = '\t',
+                   header = TRUE,
+                   stringsAsFactors = FALSE,
+                   check.names = FALSE) %>%
+    pivot_longer(names(.)[!names(.) %in% id_vars],
                  names_to = "run",
                  values_to = "intensity") %>%
-    mutate(run = sub("\\.mzML$", "", run)) %>%
+    mutate(run = str_replace(run, "\\.mzML$", "")) %>%
     rename(Run = run,
            Intensity = intensity,
            PeptideModifiedSequence = Peptide,
            ProteinName = Protein) %>%
     arrange(PeptideModifiedSequence, Intensity) %>%
-    mutate(BioReplicate = Run,
-           Condition = "unknown",
-           PrecursorCharge = 2,
+    mutate(PrecursorCharge = 2,
            IsotopeLabelType = "L",
            FragmentIon = "y0",
            ProductCharge = 1,
-           PeptideSequence = sub("[\\[\\(].*?[\\]\\)]", "", PeptideModifiedSequence)) %>%
+           PeptideSequence =
+             str_replace(PeptideModifiedSequence, "[\\[\\(].*?[\\]\\)]", "")) %>%
     select(PeptideSequence,
            PeptideModifiedSequence,
            ProteinName,
            Run,
-           BioReplicate,
-           Condition,
            Intensity,
            PrecursorCharge,
            IsotopeLabelType,
            FragmentIon,
            ProductCharge)
 
-  df
+  return(df)
+}
+
+
+# Parse the annotation file.
+annotate <- function(peptide_df, annot_csv) {
```

Comment:
Where is this annotation file coming from? Is it the input file with the added columns that we discussed earlier this week?

Reply:
Yes, exactly! The input file can optionally contain `condition` and `bioreplicate` columns, which are used here. I'm prepping another PR that greatly improves the documentation, which I'll wrap up once we have the paper together.